src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,416 @@
+    this.data = data;
+    this._defaultFormat = format;
+    this._defaultItemRenderer = itemRenderer;
+    this.keysSoFar = '';
+    this.handleFocus = this.handleFocus.bind(this);
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+    this.handleItemClick = this.handleItemClick.bind(this);
+    this.handleItemChange = function () {};
+    this.registerEvents();
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (attrName === MediaUIAttributes.MEDIA_CONTROLLER) {
+      if (oldValue) {
+        const mediaControllerEl = document.getElementById(oldValue);
+        mediaControllerEl?.unassociateElement?.(this);
+      }
+      if (newValue) {
+        const mediaControllerEl = document.getElementById(newValue);
+        mediaControllerEl?.associateElement?.(this);
+      }
+    }
+    if (attrName === 'title') {
+      this.listboxLabelElement.innerText = newValue ?? DEFAULT_LISTBOX_LABEL;
+    }
+  }
+
+  connectedCallback() {
+    this.setAttribute('role', 'button');
+    this.setAttribute('tabindex', 0);
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.associateElement?.(this);
+    }
+  }
+
+  disconnectedCallback() {
+    const mediaControllerSelector = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerSelector) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.unassociateElement?.(this);
+    }
+  }
+
+  get data() {
+    return this._data;
+  }
+
+  set data(value) {
+    if (value === this.data) return;
+    if (isNil(value) && this.data?.length === 0) return;
+    if (!(isNil(value) || isArraylike(value) || isIterable(value))) {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} data property. Must be an iterable, array-like, null, or undefined`
+      );
+      return;
+    }
+    this._data = value ?? [];
+    this.render();
+  }
+
+  get itemRenderer() {
+    return this._itemRenderer ?? this._defaultItemRenderer;
+  }
+
+  get format() {
+    return this._format ?? this._defaultFormat;
+  }
+
+  set format(value) {
+    if (value === this.format) return;
+    if (!isNil(value) || typeof value !== 'function') {
+      console.error(
+        `invalid value ${value} for ${getClassName(
+          this
+        )} format property. Must be a function, null, or undefined`
+      );
+    }
+  }
+
+  get listboxLabelElement() {
+    return this.shadowRoot.getElementById('listbox-label');
+  }
+
+  get listboxElement() {
+    return this.shadowRoot?.getElementById('listbox');
+  }
+
+  get listItemElements() {
+    return (
+      Array.from(this.listboxElement.querySelectorAll('[role="option"]')) ?? []
+    );
+  }
+
+  get activeDescendant() {
```

Comment:
`activeDescendant` is the "source of truth" for all related "active" properties, since it bottoms out in the ARIA requirements for identifying active ("focused") items via `aria-activedescendant`, defined on the listbox element itself.

Reply:
Per some investigation/callouts by @gkatsev , we may want to rely on item focus instead of `aria-activedescendant` based on sub-par support across different screen readers. I'll need to do some followup investigation, but wanted that noted here in case this comes up in discussion. Here's a very exhaustive post about this (albeit a couple of years old):
https://zellwk.com/blog/element-focus-vs-aria-activedescendant/